Source/WebCore/layout/formattingContexts/inline/InlineItemsBuilder.cpp

```diff
@@ -160,8 +160,8 @@ bool InlineItemsBuilder::traverseUntilDamaged(LayoutQueue& layoutQueue, const Bo
     }
 
     auto shouldSkipSubtree = subtreeRoot.establishesFormattingContext();
-    if (!shouldSkipSubtree && is<ElementBox>(subtreeRoot) && downcast<ElementBox>(subtreeRoot).hasChild()) {
-        auto& firstChild = *downcast<ElementBox>(subtreeRoot).firstChild();
+    if (auto* elementBox = dynamicDowncast<ElementBox>(subtreeRoot); elementBox && elementBox->hasChild() && !shouldSkipSubtree) {
+        auto& firstChild = *elementBox->firstChild();
```

Comment:
Now we unnecessarily call dynamicDowncast on plenty `shouldSkipSubtree` containers. 😿